af/fastpath/fastpath/tests/test_functional.py

```diff
@@ -121,32 +115,119 @@ def notest_core(cans):
     core(can)
 
 
-import json
 
 
-def jsha(j):
+def load_json_from_disk(fn):
+    with open(fn) as f:
+        return ujson.load(f)
+
+
+def hash(j):
     # deterministic hash to compare entries
-    x = json.dumps(j, sort_keys=True)
-    return hashlib.sha1(x.encode()).hexdigest()[:8]
+    jstr = ujson.dumps(j, sort_keys=True, ensure_ascii=False).encode()
+    return hashlib.shake_128(jstr).hexdigest(4)
 
 
 def notest_normalize_json(cans):
-    can = "testdata/2018-05-07/20180506T014008Z-CN-AS4134-web_connectivity-20180506T014010Z_AS4134_ZpxhAVt3iqCjT5bW5CfJspbqUcfO4oZfzDVjCWAu2UuVkibFsv-0.2.0-probe.json.lz4"
-    j = load_json_from_disk(can)
+    # can = "testdata/2018-05-07/20180506T014008Z-CN-AS4134-web_connectivity-20180506T014010Z_AS4134_ZpxhAVt3iqCjT5bW5CfJspbqUcfO4oZfzDVjCWAu2UuVkibFsv-0.2.0-probe.json.lz4"
+    j = load_json_from_disk(fn)
     expected = {0: "977713ee", 1: "42ec375e", 2: "9bec4a5f", 3: "a38504ed"}
     for n, entry in enumerate(j):
         esha = b"x" * 20
         entry = normalize_entry(entry, "2018-05-07", "??", esha)
         if n in expected:
-            assert jsha(entry) == expected[n]
+            assert hash(entry) == expected[n]
+
+
+# def test_normalize_yaml(cans):
+#    # data_format_version is not 0.2.0
+#    can = "testdata/2013-05-05/20130505T103213Z-VN-AS24173-http_requests-no_report_id-0.1.0-probe.yaml.lz4"
+#    #fn = "fastpath/tests/data/20190912T115043Z-ndt7-20190912T115043Z_AS15169_0B9KifXvlmX0yttm3Q5rfUL9YYed8uXrvRtPELmZuuFek7NJDf-AS15169-US-probe-0.2.0.yaml"
+#    for r in iter_yaml_lz4_reports(can):
+#        off, entry_len, esha, entry, exc = r
+#        entry = normalize_entry(entry, "2018-05-07", "??", esha)
+
+
+## YAML normalization
+
+def FIXME_test_normalize_yaml_2016(cans):
```

Comment:
What is failing here to warrant the FIXME keyword?

Reply:
(see comment above)